extensions/ql-vscode/src/databases/db-config-store.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { cloneDbConfig, DbConfig } from './db-config';
+
+export class DbConfigStore {
+  private readonly configPath: string;
+
+  private config: DbConfig;
+
+  public constructor(workspaceStoragePath: string) {
+    this.configPath = path.join(workspaceStoragePath, 'dbconfig.json');
+
+    this.config = emptyConfig;
```

Comment:
Do we also need to clone this config? If we now create multiple instances of `DbConfigStore`, they would have a reference to the same config object and would see changes made by the other instance (and the values in `emptyConfig` would also be modified).

It probably won't be a problem since we always load in a new object in `readConfig`, but I still think it's good practice to avoid having the same reference for every instance.

Reply:
That's a good shout. Updated to have a createEmptyConfig() function instead of an object.